From 7c7f56e1c6fe510a2c5e826cc49aeae3f6614f86 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Mon, 23 Jan 2017 13:36:20 +0000
Subject: [PATCH 3/3] Related: hunspell#406 use a basic_string<w_char> instead
 of vector

kcachegrind reports 1,066,887,723 -> 894,015,631 on

echo Hollo | valgrind --tool=callgrind ./src/tools/.libs/hunspell -d nl_NL
---
 src/hunspell/w_char.hxx | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/hunspell/w_char.hxx b/src/hunspell/w_char.hxx
index 84ae13c..e112b5c 100644
--- a/src/hunspell/w_char.hxx
+++ b/src/hunspell/w_char.hxx
@@ -42,7 +42,6 @@
 #define W_CHAR_HXX_
 
 #include <string>
-#include <vector>
 
 #ifndef GCC
 struct w_char {
@@ -75,7 +74,7 @@ struct replentry {
 
 namespace wide
 {
-    typedef std::vector<w_char> string;
+    typedef std::basic_string<w_char> string;
 }
 
 #endif
-- 
2.9.3

